packages/block-editor/src/hooks/color.js

```diff
@@ -276,6 +277,47 @@ export function ColorEdit( props ) {
 	);
 }
 
+/**
+ * This adds inline styles for color palette colors.
+ * Ideally, this is not needed and themes should load their palettes on the editor.
+ *
+ * @param  {Function} BlockListBlock Original component
+ * @return {Function}                Wrapped component
+ */
+export const withColorPaletteStyles = createHigherOrderComponent(
+	( BlockListBlock ) => ( props ) => {
+		const { name, attributes } = props;
+		const { backgroundColor, textColor } = attributes;
+		const { colors } = useSelect( ( select ) => {
```

Comment:
I tracked the test failure locally. I don't understand why it happens yet but it seems that commenting/removing the useSelect solves the issue. That's very weird. Any ideas why a useSelect call without impact on the rendered component would break tests here cc @aduth @epiqueras @jorgefilipecosta

Reply:
What are the symptoms of the issue? (What do you see?)

Based on what I see here, it's not really clear what issue there should be. Is it fair to say "without impact on the rendered component" if `colors` is used below in determining which props to assign?

I doubt it has any impact on fixing the issue, but a suggestion I might have here is it could be good to avoid the subscription incurred by `useSelect` until after the point where it's known whether the early `return` is reached, by creating a separate component which handles the logic after the early return and contains the hook.

I guess unless we figure that most common blocks will support colors, in which case it's probably not much an optimization.

Reply:
I agree with the proposal, it will fix the test but potentially hide an issue because the test fails even if we don't apply any style (the early return)

And I don't the symptoms are basically the test failure: it seems like something related to focus  maybe cause the "Third paragraph" is not written.

Reply:
It sounds like a race condition with the subscriptions.

Reply:
This PR is an important fix, I'm tempted to apply @aduth's fix in order to land it even if it hides the failure.

Reply:
It's not a bad idea. That's the recommended way to deal with conditional hooks, regardless of tests.

Reply:
Actually, no ideally we don't remount the block when applying a color so we shouldn't bail early here (maybe just for the support check)

Reply:
Aren't we already rerendering? For each path, a different instance of `BlockListBlock` is returned. It should still work otherwise though. Maybe worth looking at after beta 1?